docs/table.md

```diff
@@ -0,0 +1,377 @@
+          { border: undefined, backgroundColor: "#eee", content: "border:\nundefined (default)" },
+          { border: [false, false, false, true], backgroundColor: "#ddd", content: "border:\n[false, false, false, true]" },
+        ],
+        [
+          { colSpan: 2, border: true, backgroundColor: "#efe", content: "colSpan: 2\n\nborder:\ntrue" },
+        ],
+        [
+          { border: 0, backgroundColor: "#eee", content: "border:\n0 (same as false)" },
+          { border: [false, true, true, false], backgroundColor: "#ddd", content: "border:\n[false, true, true, false]" },
+        ],
+      ],
+    })
+
+![12]()
+
+---
+
+    doc.table({
+      defaultStyle: { border: false, width: 60 },
+      data: [
+        ["", "column 1", "column 2", "column 3"],
+        [
+          "row 1",
+          {
+            rowSpan: 3,
+            colSpan: 3,
+            border: true,
+            backgroundColor: "#ccc",
+            content:
+              "rowSpan: 3\ncolSpan: 3\n\nborder:\n[true, true, true, true]",
+          },
+        ],
+        ["row 2"],
+        ["row 3"],
+      ],
+    })
+
+![13]()
+
+---
+
+When defining multiple styles, the cells follow the precedence:
+
+1. `defaultStyle`
+2. `columnStyles`
+3. `rowStyles`
+4. `cellStyle`
+
+so if a table was:
+
+    doc.table({
+      defaultStyle: { border: 1 },
+      columnStyles: { border: { right: 2 } },
+      rowStyles: { border: { bottom: 3 } },
+      data: [
+        [{ border: { left: 4 } }]
+      ]
+    })
+
+The resulting cell would have a style of:
+
+    { 
+        border: { 
+            top: 1, // From the default
+            right: 2, // From the column
+            bottom: 3, // From the row
+            left: 4  // From the cell
+        }
+    }
+
+
+Internally, PDFKit keeps track of the current X and Y position of table as it
+is added to the document. This way, any calls to `text` or `table` will be placed below the table row.
+
+    doc
+        .text('before')
+        .table({
+            data: [
+                ['Column 1', 'Column 2', 'Column 3'],
+                ['One value goes here', 'Another one here', 'OK?']
+            ]
+        })
+        .text('after')
+
+![16]()
+
+## Table options
+
+- `position` - The position of the table (default `{x: doc.x, y: doc.y}`)
+- `maxWidth` - The maximum width the table can expand to (defaults to the remaining content width (offset from the tables position))
+- `columnStyles` - Column definitions of the table. (default `auto`)
+- `rowStyles` - Row definitions of the table. (default `*`)
+- `defaultStyle` - Defaults to apply to every cell
+- `data` - The data to render (not required, you can call `.row()`). This can be an iterable (async or sync)
+- `debug` - Whether to show the debug lines for all the cells (default `false`)
+
+## Cell options
+
+- `content` - The value, will be cast to a string (boolean is converted to `Y/N`, and `null` and `undefined` are not rendered but the cell is still outlined)
```

Comment:
```suggestion
- `text` - The value, will be cast to a string (boolean is converted to `Y/N`, and `null` and `undefined` are not rendered but the cell is still outlined)
```

text would make the api orthogonal with `text` method

I dont like hardcoding boolean to Y/N. What about internationalization?

Better to default to result of string casting and let user handle boolean properly

Reply:
the reason I purposefully didnt call it text was that you could pass any value in as it will be converted to a string

Reply:
The cast already occurs with `text` method, so no issue here. API consistency is more important

Reply:
renamed in latest commit